Update schema split: `train`, `model`

CBroz1 opened the pull request:
See corresponding [workflow PR](https://github.com/CBroz1/workflow-deeplabcut/pull/2). 

Has:
- Updated 2-schema split
- Updated documentation

Recent edits to functions:
- **helper funcs**: fixed bug where root_dir[0] would get appended to root_dirs whenever get_processed_dir was called.
- **extract_new_bodyparts**: added verbose arg to prevent duplicate print when calling model.Model.insert_new_model
- **insert_new_model**: because `BodyPart.extract_new_body_parts` returns a list, checking the truth value is ambiguous. Changed to 'is not None'
- **infer_ouput_dir**: remove _linking_module for VideoRecording table
- **insert_new_params**: Enforce int type on paramset_idx input. Previously failed to return when attempting to insert a duplicate bc str(x)!=int(x)

CBroz1 committed:
minor readme edit

CBroz1 committed:
Split dlc schema

CBroz1 committed:
3 -> 2 schema

CBroz1 committed:
update documentation

CBroz1 committed:
typo fix

ttngu207 commented on element_deeplabcut/model.py:
```diff
@@ -529,14 +343,11 @@ def make(self, key):
-    -> VideoRecording
-    -> Model
+    -> VideoRecording                           # Session -> Recording + File part table
+    -> model.Model                              # Must specify a DLC project_path
```
```suggestion
    -> Model                              # Must specify a DLC project_path

```

ttngu207 commented on element_deeplabcut/model.py:
```diff
@@ -529,14 +343,11 @@ def make(self, key):
-
 @schema
-class PoseEstimationTask(dj.Manual):
+class EstimationTask(dj.Manual):
```
```suggestion
class PoseEstimationTask(dj.Manual):

```
I'd still prefer the name `PoseEstimationTask` for clarity

CBroz1 replied:
Not sure why you were seeing an outdated version here after [this commit yesterday](https://github.com/CBroz1/element-deeplabcut/pull/5/commits/b84254785e42d2ae07e344bbf11bb84e2979ee51) reverted to `PoseEstimation`

ttngu207 commented on element_deeplabcut/model.py:
```diff
@@ -610,18 +422,22 @@ class BodyPartPosition(dj.Part):
         task_mode, analyze_video_params, output_dir = (
-            PoseEstimationTask & key).fetch1('task_mode', 'pose_estimation_params',
-                                             'pose_estimation_output_dir')
+            EstimationTask & key).fetch1('task_mode', 'pose_estimation_params',
```
```suggestion
            PoseEstimationTask & key).fetch1('task_mode', 'pose_estimation_params',

```

CBroz1 committed:
VideoRecording to model schema

CBroz1 committed:
update diagrams

CBroz1 committed:
See Details. Update functions for separate video tables across schemas

helper funcs: fixed bug where root_dir[0] would get appended to root_dirs whenever
              get_processed_dir was called.
extract_new_bodyparts: added verbose arg to prevent duplicate print when calling
                       model.Model.insert_new_model
insert_new_model: because `BodyPart.extract_new_body_parts` returns a list, checking the
                  truth value is ambiguous. Changed to 'is not None'
infer_ouput_dir: remove _linking_module for VideoRecording table
insert_new_params: Enforce int type on paramset_idx input. Previously failed to return
                   when attempting to insert a duplicate bc str(x)!=int(x)

ttngu207 commented on element_deeplabcut/model.py:
```diff
@@ -600,7 +434,7 @@ class PoseEstimation(dj.Computed):
         definition = """ # uses DeepLabCut h5 output for body part position
         -> master
-        -> BodyPart
+        -> model.BodyPart
```
```suggestion
        -> Model.BodyPart

```

ttngu207 commented on element_deeplabcut/model.py:
```diff
@@ -610,22 +444,26 @@ class BodyPartPosition(dj.Part):
 
         # ID model and directories
-        dlc_model = (Model & key).fetch1()
+        dlc_model = (_linking_module.model.Model & key).fetch1()
```
```suggestion
        dlc_model = (Model & key).fetch1()

```
Why the need to access `Model` via `_linking_module` and not just directly?

CBroz1 commented on element_deeplabcut/model.py:
```diff
@@ -649,7 +465,7 @@ def make(self, key):
         """
         Returns a pandas dataframe of x, y and z coordinates of the specified body_parts
-        :param key: A query specifying one PoseEstimation entry, else error is thrown.
+        :param key: A query specifying one Estimation entry, else error is thrown.
```
```suggestion
        :param key: A query specifying one PoseEstimation entry, else error is thrown.
```

CBroz1 committed:
Apply suggestions from code review

Co-authored-by: Thinh Nguyen <thinh@vathes.com>

CBroz1 committed:
remove version info from __init__.py
